ecosystem/typescript/sdk/src/token_types.ts

```diff
@@ -49,5 +47,19 @@ export interface Token {
   /** server will return string for u64 */
   amount: U64;
   /** the property map of the token */
-  token_properties: AnyObject;
+  token_properties: PropertyMap;
+}
+
+type Property = {
+  key: string;
+  value: {
+    type: string;
+    value: any;
+  };
+};
+
+export interface PropertyMap {
+  map: {
+    data: Array<Property>;
+  };
```

Comment:
is it true that PropertyMap will always look like this? Should we not be opinionated about what the PropertyMap looks like? `deserializePropertyMap` is not very useful IMO. Having `deserializeValueBasedOnTypeTag` in SDK is good enough.

Reply:
I directly modify the token class and do the conversion in the response callback now. So the user don't need to call the deserializer seprately.